### ISSUE TITLE
Is the result of endpoint `POST /session` well defined?

### ISSUE THREAD

**Issue by mercmobily** [w3c/webdriver#1201]
I am looking at the specs, especially the [POST /session](https://www.w3.org/TR/webdriver/#dfn-new-session) endpoint, used to creat ea new session.

Now... when I create a new session with the latest chromedriver, I get:

    { sessionId: '493cd4127cc2ae1b4db3ffe0610ef547',
      status: 0,
      value: 
       { acceptInsecureCerts: false,
         acceptSslCerts: false,
         applicationCacheEnabled: false,
         browserConnectionEnabled: false,
         browserName: 'chrome',
         chrome: 
          { chromedriverVersion: '2.35.528139 (47ead77cb35ad2a9a83248b292151462a66cd881)',
            userDataDir: '/tmp/.org.chromium.Chromium.TPA0jj' },
         cssSelectorsEnabled: true,
         databaseEnabled: false,
         handlesAlerts: true,
         hasTouchScreen: false,
         javascriptEnabled: true,
         locationContextEnabled: true,
         mobileEmulationEnabled: false,
         nativeEvents: true,
         networkConnectionEnabled: false,
         pageLoadStrategy: 'normal',
         platform: 'Linux',
         rotatable: false,
         setWindowRect: true,
         takesHeapSnapshot: true,
         takesScreenshot: true,
         unexpectedAlertBehaviour: '',
         version: '63.0.3239.108',
         webStorageEnabled: true } }

Whereas with Firefox I get:

    { value: 
       { sessionId: 'd9296ca9-f290-4ff0-881b-3be7138c9337',
         capabilities: 
          { acceptInsecureCerts: false,
            browserName: 'firefox',
            browserVersion: '57.0.4',
            'moz:accessibilityChecks': false,
            'moz:headless': false,
            'moz:processID': 5119,
            'moz:profile': '/tmp/rust_mozprofile.CFqlEiDUrZq9',
            'moz:webdriverClick': false,
            pageLoadStrategy: 'normal',
            platformName: 'linux',
            platformVersion: '4.10.0-42-generic',
            rotatable: false,
            timeouts: [Object] } } }

By looking at the specs, they seem to imply:

    Let body be a JSON Object initialised with:
      sessionId -  session id
      capabilities - capabilities

Basically, Firefox seems to return these two fields in the `value` property of the result, whereas chrome seems to put the sessionId in the root of the returned object, and the capablities in the `value` property.

Which one is right? Surely, the specs should be very SPECific of the return objects...?

**Comment by mercmobily** [comment]
At the moment, I am doing this in my implementation.
Please note that I want to keep the two cases separate: a W3C-confirming response, or a pot-luck search in the returned object.

Does this sound reasonable? (And, can I assume I can delete the `IF` in say 1 or 2 years, your reckon?)

      async newSession () {
        try {
          // NOTE: this.execute simply makes an HTTP call using `request` based on the method
          var res = await this.execute('post', '', { desiredCapabilities: {} })

          var isObject = (p) => typeof p === 'object' && p !== null && !Array.isAray(p)

          // W3C conforming response; checked if value is an object containing a `capabilities` object property
          // and a `sessionId` string property
          var value = res.value
          if (isObject(value) &&
              isObject(value.capabilities) &&
              typeof value.capabilities.browserName === 'string' &&
              typeof value.sessionId === 'string'
          ) {
            this.sessionCapabilities = value.capabilities
            this.sessionId = value.sessionId

          // Nevermind following W3C standards: fishing info out with pot-luck technique
          } else {
            this.sessionId = res.sessionId || value.sessionId
            if (value.browserName) this.sessionCapabilities = value
            if (value.capabilities && value.capabilities.browserName) this.sessionCapabilities = value.capabilities
            if (value.capabilities) this.sessionCapabilities = value.capabilities
          }
          if (!this.sessionId || !this.sessionCapabilities) throw new Error('Could not get sessionId and capabilities out of returned object')

          this._urlBase = `http://${this.ip}:${this.port}/session/${this.sessionId}`
          return res
        } catch (e) {
          this.sessionId = null
          this._sessionData = {}
          this._urlBase = `http://${this.ip}:${this.port}/session`
          throw (e)
        }
      }

**Comment by andreastt** [comment]
ChromeDriver doesn’t enable the WebDriver conforming behaviour by default, but I believe there is a switch to enable it somewhere. The expected behaviour is what you get from geckodriver, because it does implement the WebDriver standard by default.

**Comment by whimboo** [comment]
@mercmobily have a look here. Maybe this capability still works: https://bugs.chromium.org/p/chromedriver/issues/detail?id=1942. Otherwise I also cannot find something documented yet on the public documentation.
